File: src/main/java/com/company/erp/features/users/dto/PermissionDto.java
package com.company.erp.features.users.dto;

import com.company.erp.features.users.entity.Permission;

public record PermissionDto(Long id, String code, String description) {
    public static PermissionDto from(Permission p) {
        return new PermissionDto(p.getId(), p.getCode(), p.getDescription());
    }
}

File: app/Features/Users/Dto/PermissionDto.php
<?php

namespace App\Features\Users\Dto;

use App\Features\Users\Models\Permission;

/** Port of the Spring PermissionDto record. */
final class PermissionDto
{
    /**
     * @return array<string, mixed>
     */
    public static function from(Permission $p): array
    {
        return [
            'id' => (int) $p->id,
            'code' => $p->code,
            'description' => $p->description,
        ];
    }
}
